src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java

```diff
@@ -2522,19 +2522,96 @@ public void testLineSplitWithVoltageLevel() {
+                .body(BodyInserters.fromValue(lineAttachToAbsentLine))
+                .exchange()
+                .expectStatus().is4xxClientError()
+                .expectBody(String.class)
+                .isEqualTo(new NetworkModificationException(LINE_NOT_FOUND, "absent_line_id").getMessage());
+
+        LineAttachToVoltageLevelInfos lineAttachToVL = new LineAttachToVoltageLevelInfos("line3",
+                10.0, "AttPointId", "attPointName", vl1, "v4",
+                "1.A", attachmentLine, "nl1", "NewLine1", "nl2", "NewLine2");
+
+        webTestClient.post().uri(lineAttachUriString, TEST_NETWORK_ID)
+                .body(BodyInserters.fromValue(lineAttachToVL))
+                .exchange()
+                .expectStatus().isOk();
+
+        LineAttachToVoltageLevelInfos lineAttachToWithNewVL = new LineAttachToVoltageLevelInfos("line3",
+                10.0, "AttPointId", "attPointName", vl1, null,
+                "1.A", attachmentLine, "nl1", "NewLine1", "nl2", "NewLine2");
+
+        List<EquipmentModificationInfos> result = webTestClient.post().uri(lineAttachUriString, TEST_NETWORK_ID)
+                .body(BodyInserters.fromValue(lineAttachToWithNewVL))
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(EquipmentModificationInfos.class)
+                .returnResult().getResponseBody();
+
+        assertNotNull(result);
+        Optional<EquipmentModificationInfos> lineAttachToProper = result.stream().filter(r -> r.getType() == ModificationType.LINE_ATTACH_TO_VOLTAGE_LEVEL).findFirst();
+        assertTrue(lineAttachToProper.isPresent());
+        assertEquals(3, lineAttachToProper.get().getSubstationIds().size());
+
+        Optional<EquipmentModificationInfos> lineDeletion = result.stream().filter(r -> r.getType() == ModificationType.EQUIPMENT_DELETION).findFirst();
+        assertTrue(lineDeletion.isPresent());
+
+        LineAttachToVoltageLevelInfos lineAttachWithNewVLUpd = new LineAttachToVoltageLevelInfos("line2",
+                10.0, "AttPointId", "attPointName", vl1, null,
+                "1.A", attachmentLine, "nl1", "NewLine1", "nl2", "NewLine2");
+
+        webTestClient.put().uri("/v1/modifications/" + new UUID(128, 16) + "/line-attach-creation")
+                .body(BodyInserters.fromValue(lineAttachWithNewVLUpd))
+                .exchange()
+                .expectStatus().is4xxClientError()
+                .expectBody(String.class)
+                .isEqualTo(new NetworkModificationException(LINE_ATTACH_NOT_FOUND, "Line attach not found").getMessage());
+
+        webTestClient.put().uri("/v1/modifications/" + lineAttachToProper.get().getUuid() + "/line-attach-creation")
+                .body(BodyInserters.fromValue(lineAttachWithNewVLUpd))
+                .exchange()
+                .expectStatus().isOk();
 
```

Comment:
Maybe add test covering this type of error :
"LINE_ATTACH_ERROR(HttpStatus.INTERNAL_SERVER_ERROR)"

Reply:
Test added